xmodule/library_content_block.py

```diff
@@ -535,46 +534,26 @@ def children_are_updating(self, request, suffix=''):  # pylint: disable=unused-a
         """
         Returns whether this block is currently having its children updated from the source library.
         """
-        return Response(json.dumps(self._are_children_updating()))
-
-    def _are_children_updating(self) -> bool:
-        return (
-            self.tools.get_update_children_task_state(self.location) in
-            [UserTaskStatus.SUCCEEDED, UserTaskStatus.PENDING, UserTaskStatus.RETRYING]
+        return Response(
+            json.dumps(
+                self.tools and self.tools.are_children_updating(self)
+            )
         )
```

Comment:
To insulate this block from the details of user_tasks, I moved this logic down into LibraryTools.